nvm.sh

```diff
@@ -909,13 +909,30 @@ nvm_get_os() {
 }
 
 nvm_get_arch() {
-  local NVM_UNAME
-  NVM_UNAME="$(uname -m)"
+  local HOST_ARCH
+  local NVM_OS
+
+  NVM_OS="$(nvm_get_os)"
+  # If the OS is SunOS, first try to use pkgsrc to guess
+  # the most appropriate arch. If it's not available, use
+  # isainfo to get the instruction set supported by the
+  # kernel.
+  if [ "_$NVM_OS" = "_sunos" ]; then
+    HOST_ARCH=$(pkg_info -Q MACHINE_ARCH pkg_install)
```

Comment:
should this check `nvm_has pkg_info`? currently if it fails, it'll fall through to echoing the empty string, which isn't likely what `nvm_get_arch` should do.